dnf5/commands/module/module_list.cpp

```diff
@@ -19,13 +19,53 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 
 #include "module_list.hpp"
 
+#include <libdnf5-cli/output/modulelist.hpp>
+#include <libdnf5/module/module_query.hpp>
+
 namespace dnf5 {
 
+using namespace libdnf5::cli;
+
 void ModuleListCommand::set_argument_parser() {
     auto & cmd = *get_argument_parser_command();
     cmd.set_description("List module streams");
+
+    enabled = std::make_unique<ModuleEnabledOption>(*this);
+    disabled = std::make_unique<ModuleDisabledOption>(*this);
+    module_specs = std::make_unique<ModuleSpecArguments>(*this);
+}
+
+void ModuleListCommand::configure() {
+    auto & context = get_context();
+    context.set_load_system_repo(true);
+    context.set_load_available_repos(Context::LoadAvailableRepos::ENABLED);
 }
 
-void ModuleListCommand::run() {}
+void ModuleListCommand::run() {
+    libdnf5::module::ModuleQuery query(get_context().base, true);
+    auto module_specs_str = module_specs->get_value();
+    if (module_specs_str.size() > 0) {
+        for (const auto & module_spec : module_specs_str) {
+            for (auto & nsvcap : libdnf5::module::Nsvcap::parse(module_spec)) {
```

Comment:
I guess we could also use `const auto &` here ...

Reply:
Fixed, thanks.